x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/TextStructureUtils.java

```diff
@@ -494,6 +529,7 @@ public static Map<String, Object> makeIngestPipelineDefinition(
             if (customGrokPatternDefinitions.isEmpty() == false) {
                 grokProcessorSettings.put("pattern_definitions", customGrokPatternDefinitions);
             }
+            grokProcessorSettings.put("ecs_compatibility", ecsCompatibility);
```

Comment:
It is possible for `ecsCompatibility` to be `null` here, because `TextStructureOverrides` doesn't enforce that `ecsCompatibility` is set to something.  I think it is actually good if we explicitly set it here even if the user-supplied overrides didn't set it, because then the ingest pipeline we create will not change meaning if the default value of `ecs_compatibility` is changed in a future major release.

It would be good to think through how consistency in the values of `ecsCompatibility` can be enforced.  Maybe every class that stores it should default it to `disabled` and then enforce that it is never set to `null` after that.  Then the `writeOptionalString` and `readOptionalString` methods used with it become `writeString` and `readString`.

And this line can be

```suggestion
            grokProcessorSettings.put("ecs_compatibility", Objects.requireNonNull(ecsCompatibility));
```